### ISSUE TITLE
"Export between loop markers" causes inaccurate progress bar display

### ISSUE THREAD

**Issue by Wallacoloo** [LMMS/lmms#2138]
In the export menu, if you select "export between loop markers" and the loop markers are placed somewhere besides the beginning and end of the song, the progress bar will show inaccurate percentages whilst exporting.

I believe the issue is simply that the progress is calculated as (index of bar being exported) / (number of bars in the song), and so it discards things like which bar you started exporting from or which bar you plan to stop at.

**Comment by Wallacoloo** [comment]
Indeed, the issue is on [ProjectRenderer.cpp:176](https://github.com/LMMS/lmms/blob/f40329946024b93c92bdd9c2c46b8390a027d393/src/core/ProjectRenderer.cpp#L176)

```c++
Song::PlayPos & pp = Engine::getSong()->getPlayPos(
						Song::Mode_PlaySong );
m_progress = 0;
const int sl = ( Engine::getSong()->length() + 1 ) * 192;

while( Engine::getSong()->isExportDone() == false &&
			Engine::getSong()->isExporting() == true
						&& !m_abort )
{
	m_fileDev->processNextBuffer();
	const int nprog = pp * 100 / sl;
	if( m_progress != nprog )
	{
		m_progress = nprog;
		emit progressChanged( m_progress );
	}
}
...
Looking into seeing if there's an easy way to access the start and end export points from this function.

**Comment by Wallacoloo** [comment]
Fixed via #2139
